cloud/ec2.go

```diff
@@ -399,49 +453,54 @@ func (m *ec2Manager) SpawnHost(ctx context.Context, h *host.Host) (*host.Host, e
 	}
 	h.InstanceType = ec2Settings.InstanceType
 
-	r, err := getRegion(h)
+	region, err := getRegion(h)
 	if err != nil {
 		return nil, errors.Wrap(err, "problem getting region for host")
 	}
-	if err = m.client.Create(m.credentials, r); err != nil {
+	if err = m.client.Create(m.credentials, region); err != nil {
 		return nil, errors.Wrap(err, "error creating client")
 	}
 	defer m.client.Close()
 
 	var resources []*string
-	provider, err := m.getProvider(ctx, h, ec2Settings)
-	if err != nil {
-		msg := "error getting provider"
-		grip.Error(message.WrapError(err, message.Fields{
-			"message":       msg,
-			"host":          h.Id,
-			"host_provider": h.Distro.Provider,
-			"distro":        h.Distro.Id,
-		}))
-		return nil, errors.Wrap(err, msg)
-	}
-	if provider == onDemandProvider {
-		resources, err = m.spawnOnDemandHost(ctx, h, ec2Settings, blockDevices)
+
+	needsOnDemandHost := m.provider == onDemandProvider || h.UserHost
+
+	osName := getOsName(h)
+	var onDemandPrice float64
+	if needsOnDemandHost || m.provider == autoProvider {
+		onDemandPrice, err = pkgCachingPriceFetcher.getEC2OnDemandCost(ctx, m.client, osName, ec2Settings.InstanceType, region)
 		if err != nil {
-			msg := "error spawning on-demand host"
+			return nil, errors.Wrap(err, "error getting ec2 on-demand cost")
+		}
+	}
+
+	if !needsOnDemandHost {
+		h.Distro.Provider = evergreen.ProviderNameEc2Spot
+		resources, err = m.trySpawningSpotHost(ctx, h, ec2Settings, blockDevices, osName, onDemandPrice)
+		if m.provider == spotProvider && resources == nil {
+			if err == nil {
+				err = errors.New("spot host could not be created in any availability zones")
+			}
+			msg := "error spawning spot host"
 			grip.Error(message.WrapError(err, message.Fields{
 				"message":       msg,
 				"host":          h.Id,
 				"host_provider": h.Distro.Provider,
 				"distro":        h.Distro.Id,
 			}))
-			return nil, errors.Wrap(err, msg)
+
+			return nil, errors.Wrap(err, "failed to spawn spot host")
```

Comment:
I'd kill the log message here. logging an error and returning an error probably shouldn't happen except in some extreme cases.